spec/controllers/teachers_controller_spec.rb

```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe TeachersController, type: :controller do
+
+  describe "GET #index" do
+    it "returns http success" do
+      get :index
+      expect(response).to have_http_status(:success)
+    end
```

Comment:
We should also test that if your application is not approved, you cannot access the page. Test for the danger message and test for the redirect.